src/lseq.rs

```diff
@@ -0,0 +1,269 @@
+//! # LSEQ
+//!
+//! An LSEQ tree is a CRDT for storing sequences of data (Strings, ordered lists).
+//! It provides an efficient view of the stored sequence, with fast index, insertion and deletion
```

Comment:
Do you have the complexities for each of these ops handy? would be nice to qualify the `fast` if possible (nice to have, not necessary)

Reply:
<img width="409" alt="image" src="https://user-images.githubusercontent.com/451373/79255151-1d44ff80-7e86-11ea-91b6-521af6ccec59.png">

I'll look through all my papers to see if the author proved the bounds. Since this is fundamentally a randomized algorithm it's a little complex of course.

Intuitively the bounds for me are:
- Worst Case: 
  + insert:`O(n)`
  + delete: `O(n)`

In the worst case the identifiers are degenerate and form a list rather than a tree. I think the probability of this happening is proportional to `n / 2^n`? Not quite sure but _very_ low. LSEQ uses randomized allocation strategies at each layer of the tree to fight this kind of worst-case so it would have to be a _very_ unlucky pathological edge case.

The average case complexities for insertion and deletion should be `Θ(log n)`because we start by doing a binary search in the vector followed by identifier generation which is linear in the size of identifiers. If the tree is well-balanced the identifiers should have a size which is logarithmic in the size of the text (that's the whole point of LSEQ). Therefore the total complexity should be logarithmic on avg.

Reply:
Ah ok thanks, I was hoping it was listed in the paper and you had it handy. `log n` does sound right intuitively as the average case. 

Don't worry about doing a deeper dive on this point, it's not that important a point.

Reply:
in the paper:

L. Andre, S. Martin, G. Oster, and C.-L. Ignat, “Supporting Adaptable Granularity of Changes for Massive-scale Collaborative Editing,” in Proceedings of the 9th IEEE International Conference on Collaborative Computing: Networking, Applications and Worksharing, Austin, United States, 2013, doi: 10.4108/icst.collaboratecom.2013.254123.

I found this table:

<img width="870" alt="image" src="https://user-images.githubusercontent.com/451373/79277027-58a4f580-7ea9-11ea-81fc-fd012986e54b.png">

LSEQ should have the same complexity as LOGOOT what changes is the `i` factor. in `LOGOOT` its linear with the size of the document while `LSEQ` makes it logarithmic.